vic/drivers/cesm/src/vic_force.c

```diff
@@ -0,0 +1,303 @@
+                // CESM units: 1e-6 mol/mol
+                // VIC units: mol CO2/ mol air
+                atmos[i].Catm[j] = 1e6 * x2l_vic[i].x2l_Sa_co2prog;
+            }
+        }
+    }
+
+    if (options.LAKES) {
+        // incoming channel inflow
+        for (j = 0; j < NF; j++) {
+            for (i = 0; i < local_domain.ncells_active; i++) {
+                // CESM units: kg m-2 s-1
+                // VIC units: mm
+                atmos[i].channel_in[j] = x2l_vic[i].x2l_Flrr_flood *
+                                         global_param.snow_dt;
+            }
+        }
+    }
+
+    if (options.SNOW_BAND > 1) {
+        log_err("SNOW_BAND not implemented in vic_force()");
+    }
+    else {
+        t_offset = 0;
+    }
+
+    // Convert forcings into what we need and calculate missing ones
+    for (i = 0; i < local_domain.ncells_active; i++) {
+        for (j = 0; j < NF; j++) {
+            // vapor pressure deficit
+            atmos[i].vpd[j] = svp(atmos[i].air_temp[j]) - atmos[i].vp[j];
+            // photosynthetically active radiation
+            // TODO: Add CARBON_SW2PAR back to the parameters structure
+            // atmos[i].par[j] = param.CARBON_SW2PAR * atmos[i].shortwave[j];
+            // air density
+            atmos[i].density[j] = air_density(atmos[i].air_temp[j],
+                                              atmos[i].pressure[j]);
+            // snow flag
+            atmos[i].snowflag[j] = will_it_snow(&(atmos[i].air_temp[j]),
+                                                t_offset,
+                                                param.SNOW_MAX_SNOW_TEMP,
+                                                &(atmos[i].prec[j]), 1);
+        }
+    }
+
+
+    // Put average value in NR field
+    for (i = 0; i < local_domain.ncells_active; i++) {
+        atmos[i].air_temp[NR] = average(atmos[i].air_temp, NF);
+        // For precipitation put total
+        atmos[i].prec[NR] = average(atmos[i].prec, NF) * NF;
+        atmos[i].shortwave[NR] = average(atmos[i].shortwave, NF);
+        atmos[i].longwave[NR] = average(atmos[i].longwave, NF);
+        atmos[i].pressure[NR] = average(atmos[i].pressure, NF);
+        atmos[i].wind[NR] = average(atmos[i].wind, NF);
+        atmos[i].vp[NR] = average(atmos[i].vp, NF);
+        atmos[i].Catm[NR] = average(atmos[i].Catm, NF);
+        atmos[i].channel_in[NR] = average(atmos[i].channel_in, NF);
+        atmos[i].vpd[NR] = (svp(atmos[i].air_temp[NR]) - atmos[i].vp[NR]);
+        atmos[i].density[NR] = air_density(atmos[i].air_temp[NR],
+                                           atmos[i].pressure[NR]);
+        atmos[i].snowflag[NR] = will_it_snow(atmos[i].air_temp, t_offset,
+                                             param.SNOW_MAX_SNOW_TEMP,
+                                             atmos[i].prec, NF);
+    }
+
+    // TBD: coszen (used for some of the carbon functions)
+    // Catm (not used as far as I can tell)
```

Comment:
Again, sharing across drivers could make this issue go away as it's been handled already.  Unless this code was written pre-image-driver-sharing-PR?